astropy/io/misc/tests/test_hdf5.py

```diff
@@ -372,6 +372,50 @@ def test_preserve_meta(tmpdir):
 
 
 @pytest.mark.skipif('not HAS_H5PY')
+def test_preserve_serialized(tmpdir):
+    test_file = str(tmpdir.join('test.hdf5'))
+
+    t1 = Table()
+    t1['a'] = Column(data=[1, 2, 3], unit="s")
+    t1['a'].meta['a0'] = "A0"
+    t1['a'].meta['a1'] = {"a1": [0, 1]}
+    t1['a'].format = '7.3f'
+    t1['a'].description = 'A column'
+    t1.meta['b'] = 1
+    t1.meta['c'] = {"c0": [0, 1]}
+
+    t1.write(test_file, path='the_table', serialize_meta=True, overwrite=True)
+
+    t2 = Table.read(test_file, path='the_table')
+
+    assert t1['a'].unit == t2['a'].unit
+    assert t1['a'].format == t2['a'].format
+    assert t1['a'].description == t2['a'].description
+    assert t1['a'].meta['a0'] == t2['a'].meta['a0']
```

Comment:
I preferred to be able, if something failed, to understand exactly which meta was failing.